### ISSUE TITLE
Add cache

### ISSUE THREAD

**Issue by jperelli** [jperelli/osm-static-maps#11]
Think on how to add a directory/redis/memcache cache so that this can be used as a `<img src="...">` tag more efficiently. Maybe add a cache param like
 - cache=directory:path
 - cache=redis:url
 - cache=memcache:url
and maybe use another param like `cache_force_refresh=true` to refresh cache, or a timeout like `cache_timeout` to expire old elements saved in cache

**Comment by pascalroget** [comment]
as in - caching the binary images? If so, will probably need an _expires_ param as well

**Comment by jperelli** [comment]
Right, caching the binary images using the params (or hash of params) as key of the cache. That way only the first time the image is generated, but the next times is requested, the library just uses the cache to get and return the image fast.
